src/Message/PurchaseRequest.php

```diff
@@ -68,23 +91,28 @@ public function setSignatureMode($signatureMode)
 
     public function getData()
     {
-        $data = array();
-        $data['Ds_Merchant_Amount'] = (float)$this->getAmount();
-        $data['Ds_Merchant_Currency'] = $this->getCurrency();
-        $data['Ds_Merchant_Order'] = $this->getToken();
+        $data = [];
```

Comment:
This is for PHP 5.4+. Better keep the old way so it can be used with 5.3, as omnipay.